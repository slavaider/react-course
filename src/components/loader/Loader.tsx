import React from 'react';
import './Loader.scss';

const Loader: React.FC = () => {
  return (
    <div className="Loader">
      <div className="lds-ring">
        <div />
        <div />
        <div />
      </div>
    </div>
  );
};

export default Loader;
