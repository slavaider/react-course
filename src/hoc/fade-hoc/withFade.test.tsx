import { act, render } from '@testing-library/react';
import React from 'react';
import Loader from '../../components/loader/Loader';
import withFade from './WithFade';

test('renders correctly', () => {
  jest.useFakeTimers();
  const Hoc = withFade(Loader);
  const wrapper = render(<Hoc />);
  act(() => {
    jest.advanceTimersByTime(1000);
  });
  const loader = wrapper.getByLabelText('transition');
  expect(!loader.getAttribute('in')).toBeTruthy();
  expect(wrapper.container).toMatchSnapshot();
});
