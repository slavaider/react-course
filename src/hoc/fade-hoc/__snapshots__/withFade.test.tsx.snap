// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    aria-label="transition"
    class="wrapper my-node-enter my-node-enter-active"
  >
    <div
      class="Loader"
    >
      <div
        class="lds-ring"
      >
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
</div>
`;
